packages/coinjoin/src/client/CoinjoinPrison.ts

```diff
@@ -0,0 +1,51 @@
+export interface PrisonInmate {
+    id: string; // AccountUtxo/Alice.outpoint or AccountAddress scriptPubKey
+    sentenceStart: number;
+    sentenceEnd: number;
+    reason?: string;
+    roundId?: string;
+}
+
+export interface PrisonOptions {
+    roundId?: string;
+    reason?: string;
+    sentenceEnd?: number;
+}
+
+// Errored or currently registered inputs and addresses are sent here
+// inspiration: WalletWasabi/WabiSabi/Backend/Banning/Prison.cs
+
+export class CoinjoinPrison {
+    inmates: PrisonInmate[] = [];
+
+    detain(id: string, options: PrisonOptions = {}) {
+        const sentenceStart = Date.now();
+        const sentenceEnd = Date.now() + (options.sentenceEnd ? options.sentenceEnd : 6 * 60000);
+
+        this.inmates.push({
+            id,
+            sentenceEnd,
+            sentenceStart,
+            reason: options.reason,
+            roundId: options.roundId,
+        });
+    }
+
+    isDetained(id: string) {
+        return this.inmates.find(i => i.id === id);
+    }
+
+    // called on each status change before rounds are processed
+    release() {
+        const now = Date.now();
+        this.inmates = this.inmates.filter(inmate => inmate.sentenceEnd > now);
+    }
+
+    amnestyById(id: string) {
+        this.inmates = this.inmates.filter(inmate => inmate.id !== id);
+    }
+
+    amnestyForAll() {
+        this.inmates = [];
+    }
```

Comment:
this is never used. I'd suggest replacing unused members with a comment, something like

```
// not implemented
// - amnestyForAll
```

of course it applies only if it is not going to be use in foreseeable future

Reply:
i will remove them :) and put them back when i figure out why did i create them